internal/internal_workflow_client.go

```diff
@@ -155,98 +155,65 @@
 	workflowFunc interface{},
 	args ...interface{},
 ) (*WorkflowExecution, error) {
-	workflowID := options.ID
-	if len(workflowID) == 0 {
-		workflowID = uuid.NewRandom().String()
-	}
-
-	if options.TaskList == "" {
-		return nil, errors.New("missing TaskList")
+	startRequest, err := wc.getWorkflowStartRequest(ctx, "StartWorkflow", options, workflowFunc, args...)
+	if err != nil {
+		return nil, err
 	}
 
-	executionTimeout := common.Int32Ceil(options.ExecutionStartToCloseTimeout.Seconds())
-	if executionTimeout <= 0 {
-		return nil, errors.New("missing or invalid ExecutionStartToCloseTimeout")
-	}
+	var response *s.StartWorkflowExecutionResponse
 
-	decisionTaskTimeout := common.Int32Ceil(options.DecisionTaskStartToCloseTimeout.Seconds())
-	if decisionTaskTimeout < 0 {
-		return nil, errors.New("negative DecisionTaskStartToCloseTimeout provided")
-	}
-	if decisionTaskTimeout == 0 {
-		decisionTaskTimeout = defaultDecisionTaskTimeoutInSecs
-	}
+	// Start creating workflow request.
+	err = backoff.Retry(ctx,
+		func() error {
+			tchCtx, cancel, opt := newChannelContext(ctx, wc.featureFlags)
+			defer cancel()
 
-	// Validate type and its arguments.
-	workflowType, input, err := getValidatedWorkflowFunction(workflowFunc, args, wc.dataConverter, wc.registry)
-	if err != nil {
-		return nil, err
-	}
+			var err1 error
+			response, err1 = wc.workflowService.StartWorkflowExecution(tchCtx, startRequest, opt...)
+			return err1
+		}, createDynamicServiceRetryPolicy(ctx), isServiceTransientError)
 
-	memo, err := getWorkflowMemo(options.Memo, wc.dataConverter)
 	if err != nil {
 		return nil, err
 	}
 
-	searchAttr, err := serializeSearchAttributes(options.SearchAttributes)
-	if err != nil {
-		return nil, err
+	if wc.metricsScope != nil {
+		scope := wc.metricsScope.GetTaggedScope(tagTaskList, options.TaskList, tagWorkflowType, startRequest.WorkflowType.GetName())
+		scope.Counter(metrics.WorkflowStartCounter).Inc(1)
 	}
 
-	delayStartSeconds := common.Int32Ceil(options.DelayStart.Seconds())
-	if delayStartSeconds < 0 {
-		return nil, errors.New("Invalid DelayStart option")
+	executionInfo := &WorkflowExecution{
+		ID:    *startRequest.WorkflowId,
+		RunID: response.GetRunId(),
 	}
+	return executionInfo, nil
+}
 
-	jitterStartSeconds := common.Int32Ceil(options.JitterStart.Seconds())
-	if jitterStartSeconds < 0 {
-		return nil, errors.New("Invalid JitterStart option")
+// StartWorkflowAsync queues a workflow execution which is going to be picked up and started by Cadence backend asynchronously.
```

Comment:
Reword to match SignalWithStartWorkflowAsync?  I think that one's a bit clearer.

Reply:
I wrote this first and then a while after wrote the docs for `SignalWithStartWorkflowAsync`. Missed to update here looks like. addressing